### ISSUE TITLE
Increasing number of columns causes them to overlap

### ISSUE THREAD

**Issue by maddisondesigns** [WordPress/gutenberg#5130]
## Issue Overview
When you add more columns to the Columns block, they start overlapping

![gutenberg-multicolumns3](https://user-images.githubusercontent.com/2078245/36362961-337e9266-158d-11e8-8e86-5c46304a9f50.gif)

## Steps to Reproduce (for bugs)
1. Insert Columns block
2. Add Image block in first column
3. Click Column block and increase number of columns to 6
4. Try to insert text in columns

## Expected Behavior
Columns should never overlap

## Current Behavior
Trying to add content in columns is impossible as they overlap when the number is increased

_Firefox Quantum 58.0.2 (64-bit)
Gutenberg 2.2.0
WP 4.9.4_

**Comment by SuperGeniusZeb** [comment]
Thanks to #5658 and other recent changes, this problem has been reduced quite a bit, but there are still cases where blocks in columns are overlapping each other.
![image](https://user-images.githubusercontent.com/19592990/38430452-f4f85b76-3986-11e8-892f-7bcb44a2ed5d.png)

Seeing as that buttons wrap/break long strings of characters, I assume the same could be done for paragraphs, and thus essentially fix this issue for the most part.

It also does not help that the default content width in Gutenberg is not exactly wide, and most responsive layouts would have switched a 6-column layout to a stacked 3-column layout by the time the content width became the same as the default Gutenberg content width.

**Comment by SuperGeniusZeb** [comment]
Actually, it seems like this issue is fixed. #6049 can be considered separate from this issue, and aside from text break/wrap issues, columns appear to not be overlapping each other at all anymore. I think this issue can be closed now.

**Comment by maddisondesigns** [comment]
Nope. Columns still overlap in v2.7.0. Not only that, content extends past the edge of the block.

![gutenberg-columnsblockoverlap](https://user-images.githubusercontent.com/2078245/39458995-ea91a39a-4d3b-11e8-97e0-85463a1482cb.png)

**Comment by SuperGeniusZeb** [comment]
@maddisondesigns Wait, is this ticket about text-wrap/break issues or just columns overlapping in general?

Because I made a separate issue for text breaking/wrapping. Are you using Firefox? The text breaking/wrapping issue does not occur on Chrome/Chromium. (It is because of a difference in default CSS values between Firefox and Chrome/Chromium.) See #6049.

In the original post of this issue, images are shown overlapping other columns. This no longer happens, so I assumed that would be enough to consider this issue fixed, since #6049 covers the other issue and is more specific.

**Comment by maddisondesigns** [comment]
@SuperGeniusZeb This was originally created for the issue of text/content overlapping columns. Yep, I'm using Firefox Quantum (59.0.2 64-bit). When I re-tested this earlier today, I also noticed that content is now also extending past the edge of the block container.

The original issue wasn't specifically related to images overlapping, it was about all content overlapping. It just so happened that my example used an image to show the problem. If you look at the original gif I attached, you can see both text and images overlapping.

**Comment by SuperGeniusZeb** [comment]
@maddisondesigns So should this issue or my issue (#6049) be closed? Because they are clearly duplicates.

**Comment by GlennMartin1** [comment]
@maddisondesigns, I think this issue is resolved since 3.0.

Please confirm, and close if resolved.

**Comment by maddisondesigns** [comment]
@GlennMartin1 This is still not resolved in 3.0.1

![gutenberg_columnblockoverlap](https://user-images.githubusercontent.com/2078245/41328067-88f8e7f2-6f09-11e8-8ff7-dcc4effe88ee.png)

**Comment by SuperGeniusZeb** [comment]
@maddisondesigns it is still a duplicate of #6049 (or I guess vice-versa), though. I think that this issue should be closed as mine is focused on the remaining text issue whereas this one was originally also about images overlapping other blocks, which has now been fixed.

**Comment by maddisondesigns** [comment]
@SuperGeniusZeb This was never just about images overlapping. It was about column block content in general, overlapping other columns. I posted a screenshot back in the beginning of May, showing how content is still overlapping.

**Comment by SuperGeniusZeb** [comment]
Thanks to #7472, I think this can be closed now. :smiley:

**Comment by maddisondesigns** [comment]
@SuperGeniusZeb Nope. Column Block content still overlaps with Gutenberg 3.1.1

![gutenberg_columnsblockstilloverlaps](https://user-images.githubusercontent.com/2078245/42070978-35169216-7b9c-11e8-8401-187332f94743.png)

**Comment by SuperGeniusZeb** [comment]
@maddisondesigns Have you tested in `master`? The issue is not occurring for me there. (Also, weirdly, `master` seems to still be using the 3.1.0 as its version number.) Also, what browser are you using?

**Comment by maddisondesigns** [comment]
@SuperGeniusZeb No, this was just tested with the 3.1.1 version of the plugin from .org. I'm using Firefox Quantum 61.0 (64-bit)

**Comment by SuperGeniusZeb** [comment]
@maddisondesigns Okay, the fix came in #7472, which I guess is not in 3.1.1. If you want to test the `master` branch, you can do the following:

1. Open a terminal and change directory to some folder you want to clone the `gutenberg` repository into.
2. Use `git clone https://github.com/WordPress/gutenberg.git` (You will have to have `git` installed, of course.)
3. Once that is done, enter the newly-created `gutenberg` directory from your terminal.
4. Run `npm run package-plugin` to build the project.
5. Once this finishes, there will be a ZIP archive called `gutenberg.zip` in the `gutenberg directory`. You can upload this file to your WordPress installation. (You need to uninstall the current installation of Gutenberg on your website first, though.)

Of course, I recommend not opening up any existing posts/pages created on the release version of Gutenberg while using the `master` branch unless you want to risk them getting messed up due to block changes that get made in `master` but then reverted before an official release.

Also, once a new version comes out, you will be able to update to it just like you would if you were updating from a release version. As mentioned before, it seems that currently, the `master` branch has the wrong version number for some reason, because it is using `3.1.0` when the latest release is actually `3.1.1`; so if you install the `master` version of the plugin, the plugin will think it can be updated to a newer version... and it can, except that the newer version is actually older. :stuck_out_tongue:

**Comment by karmatosed** [comment]
I just tried to replicate this one and couldn't so for now closing.

**Comment by maddisondesigns** [comment]
@karmatosed Content doesn't seem to overlap but now it just pushes the whole editor off the page instead. I've just opened a new Issue for this (#7921)
